### PR TITLE
Global rate limit infra components

### PR DESCRIPTION
**Pull request by arkodg**
* `RateLimitInfraIR` - holds the config to manage the ratelimit service.
* `global-ratelimit` runner that subscribes to `XdsIR` messages and translates it to `RateLimitInfraIR`.
* enhance the `infrastructure` runner to subscribe to `RateLimitInfraIR` messages and translate it into platform specific ratelimit resources.

Relates to https://github.com/envoyproxy/gateway/issues/670

Signed-off-by: Arko Dasgupta <arko@tetrate.io>

### CODE REVIEW

**Comment by arkodg** [comment]
https://github.com/envoyproxy/ratelimit/pull/374 should unblock this PR and we should be able to reuse the config structure from upstream

**Comment by arkodg** [comment]
now blocked on https://github.com/envoyproxy/gateway/pull/726 which updates the Xds IR with a rate limit definiton

File: internal/globalratelimit/runner/runner.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+			} else {
+				// Publish ratelimit infra IR
+				r.RateLimitInfraIR.Store(r.Name(), result)
+			}
+		}
+	}
+	r.Logger.Info("subscriber shutting down")
+}
+
+func (r *Runner) translate(xdsIRs []*ir.Xds) (*ir.RateLimitInfra, error) {
```
[the remark this answers is not in the document]
Comment (arkodg) [reply internal/globalratelimit/runner/runner.go:70]: it doesnt need a map, it just needs to iterate across ALL xdsListeners across ALL xdsIRs

File: internal/envoygateway/config/config.go [hunk 2]
```diff
@@ -56,17 +57,26 @@ func (s *Server) Validate() error {
...
-	case s.EnvoyGateway.EnvoyGatewaySpec.Provider == nil:
+	case s.EnvoyGateway.Provider == nil:
 		return errors.New("provider is unspecified")
-	case s.EnvoyGateway.EnvoyGatewaySpec.Provider.Type != v1alpha1.ProviderTypeKubernetes:
-		return fmt.Errorf("unsupported provider %v", s.EnvoyGateway.EnvoyGatewaySpec.Provider.Type)
+	case s.EnvoyGateway.Provider.Type != v1alpha1.ProviderTypeKubernetes:
+		return fmt.Errorf("unsupported provider %v", s.EnvoyGateway.Provider.Type)
 	case len(s.Namespace) == 0:
 		return errors.New("namespace is empty string")
+	case s.EnvoyGateway.RateLimit != nil:
```
[the remark this answers is not in the document]
Comment (arkodg) [reply internal/envoygateway/config/config.go:70]: currently nothing exists for EnvoyGateway https://github.com/envoyproxy/gateway/tree/main/api/config/v1alpha1/validation
will raise a issue to track moving all this validation to `api/config/v1alpha1/validation`

**Review by danehans — APPROVED** [review]

**Pull request merged**